oat-sa/extension-tao-test-runner-plugins
# fix: Safari fullscreen alert after leaving page

Related to: [TR-4129](https://oat-sa.atlassian.net/browse/TR-4029)

The issue was happening because on opening external link Safari tab lost focus, but didn't update it's `window.outerWidth`, `window.outerHeight` and `window.innerHeight` values until tab is focused again, so I've added `focus` event listener as a solution to check for fullscreen again when tab regains focus.

## How to test

Kitchen: http://test-nfer-enot_tr-4130.playground.kitchen.it.taocloud.org:49761/

1. You can use the following test [package](https://github.com/oat-sa/extension-tao-test-runner-plugins/files/8727453/tr-4129_1652949830.zip), or create your own with an item containing external link
2. Open test in Safari
3. Go to item with external link and click it
4. Return to a tab with the test

Fullscreen prompt dialog should be displayed, user shouldn't be able to continue test in non-fullscreen mode
